app/components/search-bar.js

```diff
@@ -0,0 +1,26 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+
+export default class SearchBarComponent extends Component {
+  @service router;
+  @tracked search = this.args.search || null;
+  queryParams = ['page', 'sort', 'size', 'search'];
+
+  @action
+  handleSearch(event) {
+    this.search = event.target.value.trimStart();
+  }
+
+  @action handleKeyUp(event) {
+    if (event.code === 'Enter') {
+      this.updateSearchQuery();
+    }
+  }
+  @action
+  updateSearchQuery() {
+    let queryParams = { search: this.search };
+    this.router.transitionTo({ queryParams });
```

Comment:
make sure to reset the page to 0 when updating search

Reply:
using transitionTo works for queryparams, it's a bit unusual in this case. The usual param is to update queryParams in the controller and have emberjs take it from there. In a similar vain we could discuss whether the search bar needs to be its own component. what's the added benefit vs just having this in the template and controller?